ReadOnly and ReadAndWrite Adapter

Joe-noh opened the pull request:
This pull request tackles followings.
- `Adapter::Mock` and `Adapter::Real` are not good name for their use-case.
  - `Mock` is used when `dryrun` option is given. So `ReadOnly` is better.
  - `Real` is renamed to `ReadAndWrite`.
- Referencing `RACK_ENV` to prevent invoking `Kaname::Config.setup` on testing.
  - Using mock is straightforward way.
  - Move `Kaname::Config.setup` to adapters' `initialize`

Joe-noh committed:
restructure adapters

- make adapters name more infomative
  - Mock and Real -> ReadOnly and ReadAndWrite
- move methods that uses Yao from Resource to adapters

Joe-noh committed:
fix tests

- introduce minitest-stub_any_instance
- replace adapters name
- add some stubs

udzura commented on lib/kaname/adapter/read_only.rb:
```diff
@@ -0,0 +1,83 @@
+  module Adapter
+    class ReadOnly
+      def initialize
+        Kaname::Config.setup
```
Looks nice to move this setup process here!

udzura commented:
:+1: for me.
Any thought? @hsbt @kenchan @pyama86 @buty4649

Joe-noh committed:
make users_hash method more concise

pointed out by pyama.
https://github.com/yaocloud/kaname/commit/5df62ad04ed87d3f2a16667a0311f6dd322caa0d

Joe-noh committed:
fetch tenants inside tenant_role_hash

instead of passing as an argument

pyama86 commented:
thanks!
LGTM!!
